app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -33,8 +36,35 @@ open class FeedViewModel : ViewModel() {
         refresh()
     }
 
+    @SuppressLint("CheckResult")
     fun refresh() {
-        //todo: fix the implementation
+        NetworkAccess.feedApiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ feedResponse ->
+                handleResponse(feedResponse)
+            },{ error ->
+                handleNetworkError(error)
+            })
+    }
+
+    fun handleResponse(feedResponse: MetadataItemsList) {
+        convertFeedResponseToFeedItemList(feedResponse)
```

Comment:
We don't really need this method simply inline this to where we are using the method.